internal/services/monitor/monitor_diagnostic_setting_resource.go

```diff
@@ -53,10 +55,11 @@ func resourceMonitorDiagnosticSetting() *pluginsdk.Resource {
 			},
 
 			"target_resource_id": {
-				Type:         pluginsdk.TypeString,
-				Required:     true,
-				ForceNew:     true,
-				ValidateFunc: azure.ValidateResourceID,
+				Type:             pluginsdk.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateFunc:     azure.ValidateResourceID,
+				DiffSuppressFunc: ResourceIDCaseDifference,
```

Comment:
instead of making this case insensitive could we please fix this in the read function? we can parse the ID there/fix the case during read and save the correct case to state

Reply:
Make sense.
But the `target_resource_id` can be resource ID of different resource types, from [ms doc](https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/diagnostic-settings?tabs=portal):
> Each Azure resource requires its own diagnostic setting,

Seems we need a common method to serialize the ResourceID. Currently neither [helpers/azure/resourceid.go:13](https://github.com/hashicorp/terraform-provider-azurerm/blob/6293b0379ffe39c6f46e594c06103112f5214595/helpers/azure/resourceid.go#L13-L20) or [vendor/github.com/hashicorp/go-azure-helpers/resourcemanager/resourceids/ids.go](https://github.com/hashicorp/terraform-provider-azurerm/blob/6293b0379ffe39c6f46e594c06103112f5214595/vendor/github.com/hashicorp/go-azure-helpers/resourcemanager/resourceids/ids.go#L9-L20) supports for serialization.

Reply:
@teowa - you could do a replacement on the IDs to generally fix this without making it case insensitive by replacing `/resourcegroups/` -> `/resourceGroups/` ?